rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -0,0 +1,229 @@
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.*;
+
+
+public class RegisterBtcTransactionIT {
+
+    private final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+    private final NetworkParameters btcParams = bridgeConstants.getBtcParams();
+    private final BridgeSupportBuilder bridgeSupportBuilder = BridgeSupportBuilder.builder();
+
+    @Test
+    void whenRegisterALegacyBtcTransaction_shouldRegisterTheNewUtxoAndTransferTheRbtcBalance() {
+        // Arrange
+        ActivationConfig.ForBlock activationConfig = ActivationConfigsForTest.all().forBlock(0);
+        Repository repository = BridgeSupportTestUtil.createRepository();
+        Repository track = repository.startTracking();
+        Block rskExecutionBlock = getRskExecutionBlock();
+
+        BtcLockSenderProvider btcLockSenderProvider = new BtcLockSenderProvider();
+        FeePerKbSupport feePerKbSupport = getFeePerKbSupport();
+
+        Federation federation = P2shErpFederationBuilder.builder().build();
+        FederationStorageProvider federationStorageProvider = getFederationStorageProvider(track, federation);
+        FederationSupport federationSupport = getFederationSupport(federationStorageProvider, activationConfig, bridgeConstants.getFederationConstants());
+
+        BtcECKey btcPublicKey = new BtcECKey();
+        Coin btcTransferred = Coin.COIN;
+        BtcTransaction bitcoinTransaction = createPegInTransaction(federationSupport.getActiveFederation().getAddress(), btcTransferred, btcPublicKey);
+        TransactionOutput output = bitcoinTransaction.getOutput(0);
+        UTXO utxo = getUtxo(bitcoinTransaction, output);
+        List<UTXO> expectedFederationUtxos = Collections.singletonList(utxo);
+
+        BridgeStorageProvider bridgeStorageProvider = new BridgeStorageProvider(track, PrecompiledContracts.BRIDGE_ADDR, bridgeConstants.getBtcParams(), activationConfig);
+        BtcBlockStoreWithCache.Factory btcBlockStoreFactory = new RepositoryBtcBlockStoreWithCache.Factory(bridgeConstants.getBtcParams(), 100, 100);
+        BtcBlockStoreWithCache btcBlockStoreWithCache = btcBlockStoreFactory.newInstance(track, bridgeConstants, bridgeStorageProvider, activationConfig);
+
+        PartialMerkleTree pmtWithTransactions = createValidPmtForTransactions(Collections.singletonList(bitcoinTransaction.getHash()), bridgeConstants.getBtcParams());
+        int btcBlockWithPmtHeight = bridgeConstants.getBtcHeightWhenPegoutTxIndexActivates() + bridgeConstants.getPegoutTxIndexGracePeriodInBtcBlocks();
+        int chainHeight = btcBlockWithPmtHeight + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations();
+
+        try {
+            recreateChainFromPmt(btcBlockStoreWithCache, chainHeight, pmtWithTransactions, btcBlockWithPmtHeight, bridgeConstants.getBtcParams());
+        } catch (BlockStoreException e) {
+            fail(e.getMessage());
+        }
```

Comment:
I think this logic/block can be moved to its own method. Wdyt?

Reply:
Simply declare the method `throws Exception`. If it does the test will automatically fail.

Reply:
done